### PR TITLE
[3460] api docs improvements

### PR DESCRIPTION
**Pull request by asmega**
### Context

- https://trello.com/c/9vJWZOqy/3460-m-api-updates-before-apply-review-part-2
- This PR contains multiple improvements to the API documentation

### Changes proposed in this pull request

- Added examples to endpoints
- Added docs for sorting and filtering
- Added descriptions for schemas
- Align schema names
- Remove `?` from ends of attribute names

### Guidance to review

- Fire up docs, see README
- Should see new improved docs

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by asmega** [comment]
[[M] API Updates - Before Apply Review (part 2)](https://trello.com/c/9vJWZOqy/3460-m-api-updates-before-apply-review-part-2)

File: swagger/public_v1/component_schemas/LocationStatusAttributes.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ---
+description: "This schema is used to describe the state of a location. For example a course running a particular location may have no vacancies remaining."
 type: object
 description: "Fields relating to the state of a location."
```
Comment (defong) [inline swagger/public_v1/component_schemas/LocationStatusAttributes.yml:4]: leftovers?

File: swagger/public_v1/component_schemas/LocationStatusRelationships.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ---
+description: "This schema returns associations that belong to a location status."
 type: object
 description: "Associated data."
```
Comment (defong) [inline swagger/public_v1/component_schemas/LocationStatusRelationships.yml:4]: leftovers?

File: swagger/public_v1/component_schemas/Pagination.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+description: "This schema used to paginate through collections."
```
Comment (defong) [inline swagger/public_v1/component_schemas/Pagination.yml:2]:
```suggestion
description: "This schema used to paginate through a collection."
```

File: swagger/public_v1/template.yml [hunk 4]
```diff
@@ -20,46 +20,106 @@ components:
...
+            - 15
+            - 16
+            - 17
+            - 18
+            - 19
+            - 20
+            - 21
+            - 22
+            - 41
+            - 24
```
Comment (defong) [inline swagger/public_v1/template.yml:115]: these needs to be quote wrapped....

json generated has mixed types

File: swagger/public_v1/component_schemas/LocationStatusAttributes.yml (hunk 1 above)
Comment (asmega) [reply swagger/public_v1/component_schemas/LocationStatusAttributes.yml:4]: you make it sound like food...

File: swagger/public_v1/component_schemas/Pagination.yml (hunk 3 above)
Comment (asmega) [reply swagger/public_v1/component_schemas/Pagination.yml]: done

File: swagger/public_v1/template.yml (hunk 4 above)
Comment (asmega) [reply swagger/public_v1/template.yml]: good spot

File: swagger/public_v1/template.yml [hunk 5]
```diff
@@ -20,46 +20,106 @@ components:
     Sort:
       type: array
       example: "provider.provider_name,name"
+      description: "This schema is used to sort collections."
       items:
         type: string
         example: "name"
+        description: "A comma delimited string of fields to sort the collection by. For example `provider.provider_name,name`."
```
Comment (defong) [inline swagger/public_v1/template.yml:27]: Maybe

```suggestion
        description: "A comma delimited string of fields to sort the collection by.",
        example: "provider.provider_name,name"
```
image

File: swagger/public_v1/template.yml [hunk 6]
```diff
@@ -20,46 +20,106 @@ components:
     Sort:
       type: array
       example: "provider.provider_name,name"
+      description: "This schema is used to sort collections."
```
Comment (defong) [inline swagger/public_v1/template.yml:23]:
```suggestion
      description: "This schema is used to sort a collection."
```

File: swagger/public_v1/template.yml [hunk 7]
```diff
@@ -184,6 +252,7 @@ components:
         jsonapi:
           $ref: "#/components/schemas/JSONAPI"
     LocationStatusResource:
+      description: "This schema provides metadata about a location status."
       type: object
       description: "Wrapper object for LocationStatusAttributes."
```
Comment (defong) [inline swagger/public_v1/template.yml:257]: leftovers?


image

File: swagger/public_v1/template.yml (hunk 5 above)
Comment (asmega) [reply swagger/public_v1/template.yml]: done

File: swagger/public_v1/template.yml (hunk 7 above)
Comment (asmega) [reply swagger/public_v1/template.yml]: done

File: swagger/public_v1/template.yml (hunk 6 above)
Comment (asmega) [reply swagger/public_v1/template.yml]: done

**Comment by asmega** [comment]
The missing documentation, I'm not sure what to add for them to be honest. For example `id` is exposed but is never used but is needed to meet the JSONAPI spec

**Pull request merged**